src/flows/CostCalculator/CostCalculator.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { number, object, string } from 'yup';
+import type { InferType } from 'yup';
+import type {
+  CostCalculatorEstimateParams,
+  CostCalculatorEstimateResponse,
+  EmploymentTermType,
+} from '../../client';
+import { useForm } from 'react-hook-form';
+import { HeadlessFormOutput } from '@remoteoss/json-schema-form';
+import { Form } from '../../components/ui/form';
+
+import { Button } from '../../components/ui/button';
+import { JSONSchemaFormFields } from '../../components/form/JSONSchemaForm';
+import { SelectField } from '@/src/components/form/fields/SelectField';
+import { TextField } from '@/src/components/form/fields/TextField';
+import { useValidationFormResolver } from '@/src/components/form/yupValidationResolver';
+import {
+  useCostCalculatorCountries,
+  useCostCalcultatorEstimation,
+  useCalculatorLoadRegionFieldsSchemaForm,
+  useCompanyCurrencies,
+} from '@/src/flows/CostCalculator/hooks';
+
+const validationSchema = object({
+  country: string().required('Country is required'),
+  currency: string().required('Currency is required'),
+  region: string(),
+  salary: number()
+    .typeError('Salary must be a number')
+    .required('Salary is required'),
+});
+
+type FormValues = InferType<typeof validationSchema> & {
+  [key: string]: any;
+};
+
+type Props = {
+  title?: string;
+  includeBenefits?: boolean;
+  includeCostBreakdowns?: boolean;
+  onSubmit: (data: CostCalculatorEstimateParams) => void;
+  onSuccess: (data: CostCalculatorEstimateResponse) => void;
+  onError: (error: Error) => void;
+};
```

Comment:
I think all the props should be optional. As a consumer you might not want to run any callbacks.so I think we could update this type to:
```js
type Props = Partial<{
  title: string;
  includeBenefits: boolean;
  includeCostBreakdowns: boolean;
  onSubmit: (data: CostCalculatorEstimateParams) => void;
  onSuccess: (data: CostCalculatorEstimateResponse) => void;
  onError: (error: Error) => void;
}>;
```